### PR TITLE
Interact with Redis by str instead of bytes, migrate to SonicV2Connector with `decode_responses=True`

### PR DESCRIPTION
**Pull request by qiluo-msft**
**- What I did**
This is the following steps to migrate to SonicV2Connector with `decode_responses=True`
Please check the plan in https://github.com/Azure/sonic-py-swsssdk/pull/89

**- How I did it**

**- How to verify it**
Unit test and DUT test.

**- Description for the changelog**

### CODE REVIEW

**Review by jleveque — DISMISSED** [review]
The code is a lot easier on the eyes now :)

File: tests/mock_tables/dbconnector.py [hunk 1]
```diff
@@ -98,6 +105,20 @@ def __init__(self, *args, **kwargs):
...
+        "Return a bytestring representation of the value. Taken from redis-py connection.py"
+        if isinstance(value, bytes):
+            return value
+        elif isinstance(value, (int, long)):
+            value = str(value).encode('utf-8')
+        elif isinstance(value, float):
+            value = repr(value).encode('utf-8')
+        elif not isinstance(value, basestring):
+            value = str(value).encode('utf-8')
+        # else:
```
Comment (SuvarnaMeenakshi) [inline tests/mock_tables/dbconnector.py:118]: is this else not required?

Comment (qiluo-msft) [reply tests/mock_tables/dbconnector.py]: Resolved with complete mock.

---
In reply to: [510372790](https://github.com/Azure/sonic-snmpagent/pull/171#discussion_r510372790) [](ancestors = 510372790)

File: tests/mock_tables/dbconnector.py [hunk 2]
```diff
@@ -91,6 +91,7 @@ def __init__(self, *args, **kwargs):
         # to identify the file path to load the db json files.
         namespace = kwargs.pop('namespace')
         db_name = kwargs.pop('db_name')
+        self.decode_responses = kwargs.pop('decode_responses') == True
```
Comment (jleveque) [inline tests/mock_tables/dbconnector.py:94]: Is the `== True` necessary?

Comment (qiluo-msft) [reply tests/mock_tables/dbconnector.py:94]: Yes. Caller could provide any type, so I want to normalize it into bool to make implementation simple.

---
In reply to: [510382170](https://github.com/Azure/sonic-snmpagent/pull/171#discussion_r510382170) [](ancestors = 510382170)

**Review by jleveque — APPROVED** [review]

**Review by SuvarnaMeenakshi — APPROVED** [review]

**Pull request merged**